dlt/destinations/utils.py

```diff
@@ -292,3 +298,163 @@ def get_deterministic_temp_table_name(table_name: str, op: str) -> str:
 
     op_name = f"{table_name}_{op}"
     return f"{op_name}_{NamingConvention._compute_tag(op_name, 0.001)}"
+
+
+class WithTableReflectionAndSql(WithTableReflection, WithSqlClient):
+    pass
+
+
+def _diff_between_actual_and_dlt_schema(
+    client: WithTableReflectionAndSql,
+    schema: Schema,
+    table_name: str,
+    actual_col_names: set[str],
+    disregard_dlt_columns: bool = True,
+) -> TPartialTableSchema:
+    """Compares dlt schema with destination table schema and returns columns that appear to be missing.
+
+    This function identifies columns that exist in the dlt schema but are missing from the actual
+    destination table. It's used during schema synchronization to detect when columns may have
+    been dropped from the destination and need to be removed from the dlt schema as well.
+
+    However, dlt internal columns (_dlt_id, _dlt_load_id) are treated specially because:
+
+    1. Users rarely drop dlt internal columns manually, and if they did,
+       dlt cannot recover from this situation anyway.
+
+    2. Athena has a constraint where dlt columns exist in the data but not in the table metadata:
```

Comment:
this looks like a bug! or columns starting with `_` are not visible in INFORMATION SCHEMA?